hera_cal/apply_cal.py

```diff
@@ -383,10 +386,20 @@ def apply_cal(data_infilename, data_outfilename, new_calibration, old_calibratio
                 # update redundant data. Don't partial write.
                 hd_red.update(nsamples=data_nsamples, flags=data_flags, data=data)
             else:
-                # partial write works for no redundant averaging.
-                hd.partial_write(data_outfilename, inplace=True, clobber=clobber, add_to_history=add_to_history, **kwargs)
+                if vis_units is None:
+                    if hasattr(hc, 'gain_scale') and hc.gain_scale is not None:
+                        vis_units = hc.gain_scale
```

Comment:
Again, I'd suggest issuing a warning if this changes the units.

Reply:
Let me know if it makes sense to not issue a warning when `vis_units="UNCALIB"`. This is the default calibration scenario so it would seem strange issue a warning if changing units from `"UNCALIB"`.

Reply:
I'm fine not issuing a warning in that case.

Reply:
I've added this when hd.vis_units is not None and hd.vis_units != 'uncalib'